arch/posix/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+zephyr_cc_option_ifdef(CONFIG_LTO -flto)
+
+zephyr_ld_option_ifdef(CONFIG_COVERAGE
+	-lgcov
+	)
```

Comment:
When you use zephyr_ld_option the build system will silently omit -lgcov if the toolchain does not support it.

I assume that is not desired behaviour and that you want to use

zephyr_link_libraries(-lgcov)

instead.

Reply:
=> @nashif